File: app/src/main/java/com/lakooz/lpctest/networking/IRestApiClient.java
package com.lakooz.lpctest.networking;

import com.lakooz.lpctest.model.Pot;

import java.util.List;

import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.POST;

interface IRestApiClient {


    @GET("/get-pots")
    fun getData() : Single<List<Pot>>


    @POST("/create-pot")
    fun createPot() : Single<Pot>
}

File: app/src/main/java/com/lakooz/lpctest/networking/IRestApiClient.kt
package com.lakooz.lpctest.networking

import com.lakooz.lpctest.model.Pot
import io.reactivex.Single
import retrofit2.http.*

interface IRestApiClient {

    @GET("get-pots")
    fun getData() : Single<List<Pot>>

    @FormUrlEncoded
    @POST("create-pot")
    fun createPot(@Field("category") category : Int) : Single<Pot>
}